app/src/main/java/com/example/dispatcher/presentation/main/view/MainActivity.kt

```diff
@@ -1,29 +1,37 @@
-package com.example.dispatcher.common.base
+package com.example.dispatcher.presentation.main.view
 
 import android.os.Bundle
+import android.view.View
+import androidx.activity.viewModels
 import androidx.appcompat.app.AppCompatActivity
+import androidx.core.view.GravityCompat
+import androidx.drawerlayout.widget.DrawerLayout
 import androidx.fragment.app.Fragment
 import com.example.dispatcher.R
 import com.example.dispatcher.common.utils.FirebaseCrashlyticsManager
 import com.example.dispatcher.databinding.ActivityMainBinding
 import com.example.dispatcher.presentation.favorites.view.FavoritesFragment
 import com.example.dispatcher.presentation.homepage.view.HomeFragment
+import com.example.dispatcher.presentation.main.MainViewModel
 import com.example.dispatcher.presentation.profile.view.ProfileFragment
+import com.example.dispatcher.presentation.search.view.SearchFragment
 
 class MainActivity : AppCompatActivity() {
 
     private lateinit var binding: ActivityMainBinding
+    private val mainViewModel: MainViewModel by viewModels()
+    private lateinit var drawerLayout: DrawerLayout
+
 
     override fun onCreate(savedInstanceState: Bundle?) {
```

Comment:
arrange this function better, extract the buttons listeners to function, make it cleaner